pybricks/_common.py

```diff
@@ -404,6 +404,25 @@ def off(self):
         """Turns off the light."""
         pass
 
+    def pattern(self, pattern, duration, repeat=True):
+        """Make the light brightness follow a pattern as a function of time.
+
+        The specified pattern function will be sampled at 64 points between 0
+        and the specified duration. The light will be held at a constant
+        brightness between samples.
+
+        This function is not blocking. The pattern will be shown as the
+        user program continues.
+
+        Arguments:
+            pattern (callable): Function of the
+                form ``b = func(t)``` that returns the brightness ``b``
+                of the (0--100.0) as a function of time ``t`` in milliseconds.
+            time (:ref:`time`): Duration of the pattern.
+            repeat (bool): Whether to keep repeating the pattern after it
```

Comment:
Either back to sys or off -- or maybe let's keep things simple, drop the argument and just always repeat.

Reply:
always repeating sounds good

Reply:
Some backgrond thoughts regarding the bigger picture: this is secretly the precursor to generic smooth motor patterns (generalizing the default trapezoidal speed pattern we have today). In that case, we may not only have repeat but also wrap, so it repeats but starts where it left off.

So leaving off the argument so we have more time to think about it could be nice.

Reply:
> always repeating sounds good

ack, I'll update.